binding.cc

```diff
@@ -390,8 +390,7 @@ namespace zmq {
 
   void
   Socket::UV_PollCallback(uv_poll_t* handle, int status, int events) {
-    assert(status == 0);
-
+    if (status != 0) NanThrowRangeError("I/O status: socket not ready !=0 ");
```

Comment:
status is indeed an argument, but since it is in a callback and not supplied by the user, using range error is a bit strange. I would make the assumption that libuv is always correct when calling the callback, thus, if the status argument is bad, it is because something else caused an error.

To summarize: `RangeError` ought to be used when an argument **supplied from the outside** lies beyond a certain range.